showkase-component-impressions/src/main/java/com/vinaygaba/showkase_component_impressions/ImpressionCollector.kt

```diff
@@ -0,0 +1,55 @@
+package com.vinaygaba.showkase_component_impressions
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
+
+internal class ImpressionCollector<T>(
+    val coroutineScope: CoroutineScope,
+    val onVisibilityEvent: (ShowkaseVisibilityEvent<T>) -> Unit
+) {
+    private val mutableImpressionEvents = MutableStateFlow<ShowkaseVisibilityEvent<T>?>(null)
+
+    @OptIn(FlowPreview::class)
+    val impressionEvents: Flow<ShowkaseVisibilityEvent<T>> =
```

Comment:
the `debounce(200)` is an interesting decision to hide as an implementation detail. maybe it should be exposed as a parameter to the modifier and be optional?

Reply:
This was added because I noticed that onGlobalPositioned often publishes multiple identical events and so debouncing was important. I figured this didn't need to be exposed by the library and choosing a reasonable default was enough as its an implementation detail for the end user that they don't need to be aware about. Do you think there's a use case where having control over this would be valuable?